spec/haskell/src/SEL4/Object/Structures.lhs

```diff
@@ -501,12 +501,12 @@ Various operations on the free index of an Untyped cap.
 
 > type TcbFlags = Word
 
-> data TcbFlag
->     = NoFlag
->     | ArchFlag ArchTcbFlag
+> noFlag :: TcbFlags
+> noFlag = 0x0
```

Comment:
noFlag seems unused, but I guess it has applicability in generic proofs later?

The TcbFlag/TcbFlags distiction is not immediately obvious, perhaps a comment above `TcbFlags` beyond it being a `Word`? Did I miss comments explaining any of this?

Reply:
Yeah, `noFlag` isn't currently used but I can imagine uses for it in the future and it better matches the C so I thought it best to include it now.

I've added a comment for `TcbFlags`, let me know if it makes sense and helps.